x-pack/legacy/plugins/task_manager/task_store.ts

```diff
@@ -137,6 +138,39 @@ export class TaskStore {
     return savedObjectToConcreteTaskInstance(savedObject);
   }
 
+  /**
+   * Reschedules a previously scheduled task.
+   *
+   * @param task - The task being rescheduled.
+   */
+  public async reschedule(
+    taskInstanceScheduling: TaskInstanceScheduling
+  ): Promise<ConcreteTaskInstance> {
+    const taskInstance = await this.getTask(taskInstanceScheduling.id);
+    return await this.update(
+      applyConcreteTaskInstanceDefaults(
+        taskInstance.status === 'idle'
+          ? {
+              ...omit(taskInstance, 'runAt'),
+              ...taskInstanceScheduling,
+            }
+          : {
+              ...taskInstance,
+              ...omit(taskInstanceScheduling, 'runAt'),
+            }
+      )
+    );
+  }
+
+  /**
+   * Get a task by id
+   *
+   * @param task - The task being scheduled.
+   */
+  public async getTask(id: string): Promise<ConcreteTaskInstance> {
```

Comment:
nit: For consistency, we can call this function `get`.

Reply:
So, it *was* `get` at first and I renamed it because I felt that that would help differentiate between `fetch` and `search`, making it explicit that this is a _single task_, while the others are _multiples_.... I'd consider renaming `fetch` to `fetchTasks` infact...